### PR TITLE
ZIO 2.0 Test Reporting

### PR DESCRIPTION
**Pull request by hmemcpy**
This PR overhauls the test reporting to work together with the new async test output. It re-introduces the built-in IntelliJ reporting, as well as fixes the summary rendering.

Summary:

* Removes `DefaultTestReporter`. The reporting is now done by the `ZTestEventHandler`, which is responsible for producing test events that are *rendered* by the specified Test Renderer (console or IntelliJ)
* Simplifies producing the rendered output by emitting a data structure that holds the test results
* Restores printing the summary of the executed specs!

### CODE REVIEW

File: test-sbt/shared/src/main/scala/zio/test/sbt/BaseTestTask.scala [hunk 1]
```diff
@@ -18,12 +18,11 @@ abstract class BaseTestTask[T](
   val runtime: zio.Runtime[T]
 ) extends Task {
 
-  protected def sharedFilledTestlayer(
-    console: Console
-  )(implicit trace: Trace): ZLayer[Any, Nothing, TestEnvironment with TestLogger with ZIOAppArgs with Scope] = {
+  protected def sharedFilledTestLayer(implicit
```
Comment (swoogles) [inline test-sbt/shared/src/main/scala/zio/test/sbt/BaseTestTask.scala:21]: Thanks for cleaning up the layer!

File: test/shared/src/main/scala/zio/test/TestOutput.scala [hunk 2]
```diff
@@ -116,12 +116,12 @@ object TestOutput {
...
                for {
                  currentOutput <- getAndRemoveSectionOutput(reporterEvent.id)
-                 _             <- printToConsole(currentOutput)
+                 _             <- print(currentOutput)
                } yield ()
              )
       } yield ()
 
-    private def printToConsole(events: Chunk[ExecutionEvent]) =
+    private def print(events: Chunk[ExecutionEvent]) =
```
Comment (swoogles) [inline test/shared/src/main/scala/zio/test/TestOutput.scala:124]: Yup, better name

**Comment by hmemcpy** [comment]
@swoogles thank you for reviewing! I'm currently 40k feet in the air, will fix the failing test for Scala 3 and any other comments you might have!

**Review by swoogles — DISMISSED** [review]

**Comment by swoogles** [comment]
New features and improved several bits of the old code, thanks for taking care of this!

**Review by adamgfraser — APPROVED** [review]

**Pull request merged**